sweepbatcher: add option WithPublishErrorHandler

starius opened the pull request:
**WithPublishErrorHandler** sets the callback used to handle publish errors.

It can be used to filter out noisy messages.


#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

bhandras commented on sweepbatcher/sweep_batcher.go:
```diff
@@ -159,6 +160,15 @@ type VerifySchnorrSig func(pubKey *btcec.PublicKey, hash, sig []byte) error
+// logAllErrorsAsWarnings is an instance of PublishErrorHandler which logs all
+// errors as warnings.
+func logAllErrorsAsWarnings(err error, errMsg string, log btclog.Logger) {
+	log.Warnf("%s: %v", errMsg, err)
```
nit: I think the default could be a little bit smarter too, like it could filter out obvious non-errors such as when it fails to RBF due to low fee, etc.

starius replied:
I updated the function. Now it logs "insufficient fee" errors as info, everything else as a warning.

bhandras reviewed (approved):
LGTM, this is a neat idea! 🎉

hieblmi commented on sweepbatcher/sweep_batcher_test.go:
```diff
@@ -541,6 +541,122 @@ func testTxLabeler(t *testing.T, store testStore,
+	<-lnd.RegisterSpendChannel
+
+	// The first attempt to publish the batch tx is expected to fail.
+	require.ErrorIs(t, <-publishErrorChan, testPublishError)
```
👌

hieblmi reviewed (approved):
LGTM 💾

starius commented:
I added commit "go.mod: update btcwallet" to have `chain.ErrInsufficientFee` symbol.
That required LND version update. I updated to v0.18.3-beta.rc1 in a separate commit.
